server.js

```diff
@@ -3,36 +3,260 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+import booksData from './data/books.json'
+
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/book-project"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Book = mongoose.model('Book', {
+  bookID: {
+    type: Number
+  },
+  title: {
+    type: String
+  },
+  authors: {
+    type: String
+  },
+  average_rating: {
+    type: Number
+  },
+  isbn: {
+    type: String
+  },
+  isbn13: {
+    type: String
+  },
+  language_code: {
+    type: String,
+  },
+  num_pages: {
+    type: Number
+  },
+  ratings_count: {
+    type: Number
+  },
+  text_reviews_count: {
+    type: Number
+  }
+})
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany()
+    booksData.forEach(async (book) => await new Book(book).save())
+  }
+  seedDatabase()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
+const myEndPoints = require('express-list-endpoints')
+const ERROR = { error: 'Data not found'}
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+//handling data connection error
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable'})
+  }
+})
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+    
+  
+  if(!res) {
+    res
+    .status(404)
+    .send({ error: 'Oops! Something goes wrong. Try again later!'})
+  }
+  res.send(myEndPoints(app))
+})
+//API endpoints for books with query options for author, title; otherwise, display 20 data per page (sorted by average_rating) and allow user to request for more pages
+app.get('/books', async (req, res) => {
+  const { title, author } = req.query
```

Comment:
One tip: you can use the following line of code.  This is a clean way of setting queries. This allows all database fields to be queries
`const allBooks = await Book.find(req.query)`